### ISSUE TITLE
[shipit-static-analysis] Off-by-one error in targeted revisions

### ISSUE THREAD

**Issue by jankeromnes** [mozilla-releng/services#351]
Recently, this review request was posted: https://reviewboard.mozilla.org/r/143884/ (it includes two commits to be reviewed).

However, when I run `./static_analysis_shared.sh 0c5dba81e363681b0488995912dc8f35a4fd11db 3bb3d47c37232ecf32eeb4ad7fc278779223aa9c`, I observe the following logs:

```
[nix-shell:/services/src/shipit_static_analysis]$ ./static_analysis_shared.sh 0c5dba81e363681b0488995912dc8f35a4fd11db 3bb3d47c37232ecf32eeb4ad7fc278779223aa9c
shipit_static_analysis.workflow: Clone mozilla central (dir='/tmp/cache_static_analysis/static-analysis')
shipit_static_analysis.workflow: Pull from review (revision='0c5dba81e363681b0488995912dc8f35a4fd11db')
shipit_static_analysis.workflow: Modified files (files=['security/manager/ssl/StaticHPKPins.h'])
shipit_static_analysis.workflow: Mach configure ...
shipit_static_analysis.workflow: Mach build backend ...
shipit_static_analysis.workflow: Mach build exports ...
shipit_static_analysis.workflow: Clang output (output=b'Enabled checks:\n    modernize-loop-convert\n    modernize-raw-string-literal\n    modernize-use-auto\n    modernize-use-bool-literals\n    modernize-use-default\n    modernize-use-nullptr\n    modernize-use-override\n\n\n')
shipit_static_analysis.workflow: Pull from review (revision='3bb3d47c37232ecf32eeb4ad7fc278779223aa9c')
shipit_static_analysis.workflow: Modified files (files=['devtools/client/framework/devtools.js', 'devtools/shim/DevToolsShim.jsm'])
shipit_static_analysis.workflow: Mach configure ...
shipit_static_analysis.workflow: Mach build backend ...
shipit_static_analysis.workflow: Mach build exports ...
shipit_static_analysis.workflow: Clang output (output=b'Enabled checks:\n    modernize-loop-convert\n    modernize-raw-string-literal\n    modernize-use-auto\n    modernize-use-bool-literals\n    modernize-use-default\n    modernize-use-nullptr\n    modernize-use-override\n\n\n')
```

While clang-tidy static analysis doesn't make sense for JS commits (rendering this task run pointless), you can observe that the list of modified files is wrong. It should be

```
shipit_static_analysis.workflow: Pull from review (revision='0c5dba81e363681b0488995912dc8f35a4fd11db')
shipit_static_analysis.workflow: Modified files (files=['devtools/client/framework/devtools.js', 'devtools/shim/DevToolsShim.jsm'])
[...]
shipit_static_analysis.workflow: Pull from review (revision='3bb3d47c37232ecf32eeb4ad7fc278779223aa9c')
shipit_static_analysis.workflow: Modified files (files=['addon-sdk/source/lib/dev/debuggee.js', 'addon-sdk/source/lib/dev/toolbox.js', 'addon-sdk/source/lib/dev/utils.js','addon-sdk/source/lib/sdk/addon/runner.js', 'addon-sdk/source/test/addons/page-mod-debugger-post/main.js', 'addon-sdk/source/test/addons/page-mod-debugger-pre/main.js'])
```

And given that the **second** commit shows the correct list for the **first** commit, I assume that there is an off-by-one error in the list of modified files (the first commit probably shows which file a commit from the previous request did modify).

**Comment by jankeromnes** [comment]
Actually, when looking at which files were modified by https://reviewboard-hg.mozilla.org/gecko/rev/86e8100df861 (changeset 586071), it appears that the list from https://reviewboard-hg.mozilla.org/gecko/rev/fe7c99424587 (changeset 586067) was shown: `shipit_static_analysis.workflow: Modified files (files=['addon-sdk/source/lib/dev/debuggee.js', 'addon-sdk/source/lib/dev/toolbox.js', 'addon-sdk/source/lib/dev/utils.js', 'addon-sdk/source/lib/sdk/addon/runner.js', 'addon-sdk/source/test/addons/page-mod-debugger-post/main.js', 'addon-sdk/source/test/addons/page-mod-debugger-pre/main.js'])`. What I find puzzling is that there doesn't appear to be a direct relationship between both revisions (neither is the other's parent).

**Comment by jankeromnes** [comment]
@La0 found that using `parent:revision` instead of `revision:tip` (with something like https://github.com/La0/mozilla-relengapi/commit/d8fdd2a7547b96821d150fa7262d64cdd7502ca0#diff-cd6247ae249a1173f4bc307666753226R72) seems to yield the correct set of changed files.

**Comment by jankeromnes** [comment]
Fixed by 44c8ba750ee669b41b52f31a7982df320718c923. Thanks!
